code/game/gamemodes/miniantags/revenant/revenant_abilities.dm

```diff
@@ -384,3 +385,77 @@
 /obj/machinery/light/defile()
 	flicker(30)
 
+/**
+ * Makes objects be haunted and then throws them at people, spooky!
+ */
+/obj/effect/proc_holder/spell/aoe_turf/revenant/haunt_object
+	name = "Haunt Objects"
+	desc = "Empower nearby objects to you with ghostly energy, causing them to attack nearby mortals. \
+		Items closer to you are more likely to be haunted."
+	action_icon_state = "haunt"
+	var/max_targets = 7
+	var/aoe_radius = 5
+	var/list/attack_timer = list()
+	base_cooldown = 60 SECONDS
+	unlock_amount = 150
+	cast_amount = 50
+	stun = 3 SECONDS
+	reveal = 10 SECONDS
+
+/obj/effect/proc_holder/spell/aoe_turf/revenant/haunt_object/create_new_targeting()
+	var/datum/spell_targeting/aoe/turf/T = new()
+	T.range = aoe_radius
+	return T
+
+/obj/effect/proc_holder/spell/aoe_turf/revenant/haunt_object/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+	if(!attempt_cast(user))
+		return
+	var/successes = 0
+	for(var/turf/T in targets)
+		for(var/obj/item/nearby_item in T)
+			// Don't throw around anchored things or dense things
+			// (Or things not on a turf but I am not sure if range can catch that)
+			if(nearby_item.anchored || nearby_item.density || !isturf(nearby_item.loc))
+				continue
+			// Don't throw abstract things
+			if(nearby_item.flags & ABSTRACT)
+				continue
+			// Don't throw things we can't see
+			if(nearby_item.invisibility >= INVISIBILITY_REVENANT)
+				continue
+
+			var/distance_from_user = get_dist(get_turf(nearby_item), get_turf(user))
+			var/chance_of_haunting = 150 * (1 / distance_from_user)
+			if(!prob(chance_of_haunting))
+				continue
+
+			if(successes >= max_targets)
+				return
+
+			INVOKE_ASYNC(src, .proc/make_spooky, nearby_item, user)
```

Comment:
Why the `INVOKE_ASYNC` here? I don't see a direct/indirect sleep in `make_spooky` but I might have missed it